### PR TITLE
Update gringo to version 4.5.0

### PR DESCRIPTION
**Pull request by illenseer**
* Update gringo to version 4.5.0
* Add new reify tool
* Fix build problem: https://sourceforge.net/p/potassco/bugs/104/
* Update aspcud to 1.9.1 for gringo 4.5.0

### CODE REVIEW

File: Library/Formula/aspcud.rb [hunk 1]
```diff
@@ -3,15 +3,15 @@
...
+  url 'https://downloads.sourceforge.net/project/potassco/aspcud/1.9.1/aspcud-1.9.1-source.tar.gz'
+  sha1 'db1ce85814699fb77aff41f4f6df0721d6851756'
 
-  bottle do
-    revision 1
-    sha1 "9d08bb4dfab9afd90b0ca3b3b3f48733869670a6" => :mavericks
-    sha1 "dc6c376297ce949034d67b4f8760b67427e9d60b" => :mountain_lion
-    sha1 "705c19c367ee0740eaa0d8130e0619d0879d7db5" => :lion
-  end
+  # bottle do
```
Comment (DomT4) [inline Library/Formula/aspcud.rb]: Please leave this block how it was. We'll update it when we pull.

File: Library/Formula/aspcud.rb [hunk 2]
```diff
@@ -3,15 +3,15 @@
 class Aspcud < Formula
   desc "Package dependency solver"
   homepage 'http://potassco.sourceforge.net/'
-  url 'https://downloads.sourceforge.net/project/potassco/aspcud/1.9.0/aspcud-1.9.0-source.tar.gz'
-  sha1 'ae77772c2424620b3064d0dfe795c26b1c8aa778'
+  url 'https://downloads.sourceforge.net/project/potassco/aspcud/1.9.1/aspcud-1.9.1-source.tar.gz'
+  sha1 'db1ce85814699fb77aff41f4f6df0721d6851756'
```
Comment (DomT4) [inline Library/Formula/aspcud.rb]: Please use `sha256` here.

File: Library/Formula/gringo.rb [hunk 3]
```diff
@@ -3,16 +3,16 @@
 class Gringo < Formula
   desc "Grounder to translate user-provided logic programs"
   homepage 'http://potassco.sourceforge.net/'
-  url 'https://downloads.sourceforge.net/project/potassco/gringo/4.4.0/gringo-4.4.0-source.tar.gz'
-  sha1 'c39a1c3cfe64b62e39e6abcc8f813e2d1d17251e'
+  url 'https://downloads.sourceforge.net/project/potassco/gringo/4.5.0/gringo-4.5.0-source.tar.gz'
+  sha1 'bbe67ed1d5c6f5ca6d5958e04b706595d39b4653'
```
Comment (DomT4) [inline Library/Formula/gringo.rb]: Same comments about `sha256` and not editing the `bottle do` block apply here as well.

File: Library/Formula/gringo.rb [hunk 4]
```diff
@@ -3,16 +3,16 @@
...
   revision 1
```
Comment (DomT4) [inline Library/Formula/gringo.rb]: `revision`s need to be removed on version bumps - They get reset to 0.

File: Library/Formula/gringo.rb [hunk 5]
```diff
@@ -26,7 +26,10 @@ def install
     inreplace "SConstruct",
               "env['CXX']            = 'g++'",
               "env['CXX']            = '#{ENV.cxx}'"
-    scons "--build-dir=release", "gringo", "clingo"
-    bin.install "build/release/gringo", "build/release/clingo"
+    inreplace "libclasp/src/clasp_output.cpp",
```
Comment (DomT4) [inline Library/Formula/gringo.rb]: Is this something that needs reporting upstream?

Comment (illenseer) [inline Library/Formula/gringo.rb]: The bug is known upstream. http://sourceforge.net/p/potassco/bugs/104/

**Comment by illenseer** [comment]
@DomT4: Thank you for guidance, I hope everything is fine now.

File: Library/Formula/gringo.rb [hunk 6]
```diff
@@ -26,7 +25,12 @@ def install
     inreplace "SConstruct",
               "env['CXX']            = 'g++'",
               "env['CXX']            = '#{ENV.cxx}'"
-    scons "--build-dir=release", "gringo", "clingo"
-    bin.install "build/release/gringo", "build/release/clingo"
+    # Fix build problems
+    # https://sourceforge.net/p/potassco/bugs/104/
+    inreplace "libclasp/src/clasp_output.cpp",
+              "	using std::isnan;",
```
Comment (DomT4) [inline Library/Formula/gringo.rb]: Does the `inreplace` break without the spacing at the start, out of curiosity?

Comment (illenseer) [inline Library/Formula/gringo.rb]: I tried it without and it doesn't break anything. I could remove the spacing, if desired.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/889aeead054f3680f6ab0f8ff600e538cd66612b & https://github.com/Homebrew/homebrew/commit/a0b94fc38fb1f4d04d9f3f15afcea12771c34587. Thank you for your contribution to Homebrew, we appreciate it! :smiley_cat:.

Please note ideally the commit message here for version bumps is a simple `gringo 4.5.0` or similar, Thanks!

**Pull request closed without merging**